crypto: Add nrf_cc3xx v0.9.16 (#850)
nrfconnect/sdk-nrfxlib: crypto/nrf_cc310_mbedcrypto/CHANGELOG.rst

All notable changes to this project are documented in this file.

nrf_cc3xx_mbedcrypto - 0.9.16
*****************************

New version of the runtime library with the following bug fixes and improvements:

> * Built with MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER enabled to ensure support with and without TF-M.

Reviewer: Ensure what support?

Was nrf_cc3xx_mbedcrypto not supported to be used with TF-M before?

Author: Fixed

> * Built with MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER enabled to ensure support with and without TF-M.

Reviewer: ```suggestion
* Now built with MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER enabled to ensure support with and without TF-M.
```

Author: Fixed

> * Now built with MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER enabled to ensure that we can support PSA key type with owner ID (for TF-M builds) and without owner ID (without TF-M enabled) from a single library.

Reviewer 2: ```suggestion
* The library is now built with MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER enabled to ensure that PSA key type with owner ID (for TF-M builds) and without owner ID (without TF-M enabled) can be supported from a single library.
```

* Added support for zero input message length for EDDSA for RFC test compliance.

> * Removed unused trace functions from runtime library.

Reviewer: ```suggestion
* Removed unused trace functions.
```

Reviewer: Specifying runtime library sounds like it's being removed from one part of, or a variant of nrf_cc3xx_mbedcrypto. But by runtime library, one is referring to nrf_cc3xx_mbedcrypto, right?

Author: runtime library is already mentioned a few lines above, so the information can be removed. Will fix

Author: Fixed

> * Added support for zero input message length for EDDSA for RFC test compliance.

Reviewer 2: ```suggestion
* Added support for zero input message length for EdDSA for RFC test compliance.
```

> Library built against Mbed TLS version 3.1.0.
> 

Reviewer: ```suggestion
```

This is not a change.

Author: This is not a change statement

Author: The changelog-items wil be handled in a subsequent PR. We must prioritize adding this new library. 

@SebastianBoe: Please add  a JIRA-ref in the comment and resolve, so we can get this in

> This version is dependent on the nrf_cc310_platform or nrf_cc312_platform library for low-level initialization of the system and proper RTOS integration.
> 

Reviewer: ```suggestion
```

This is not a change either.

Author: This is not a change statement

Reviewer: Then it shouldn't be in the changelog.

Author: No. It is an essential note about the runtime-library and it hurts nobody to mention it

Reviewer: If it is essential information then it should go in the documentation, not the changelog.

The negative consequences of having it in the changelog is that it is being communicated that this has changed, and also we are bloating the changelog with documentation that is constant for every release.

Author: Won't change in this PR, discussions to be had offline on the topic of documentation and changelog
